src/main/java/lumina/main/Lumina.java

```diff
@@ -0,0 +1,183 @@
+package lumina.main;
+
+import lumina.exception.LuminaException;
+import lumina.parser.Parser;
+import lumina.storage.Storage;
+import lumina.task.TaskList;
+import lumina.ui.Ui;
+
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a simple chatbot.
+ * It uses predefined set of responses and can be extended for more complex behaviour
+ */
+public class Lumina {
+
+    // counter for instances
+    private static int luminaCount = 0;
+
+    // composed objects
+    private Ui ui;
+    private Parser parser;
+    private Storage storage;
+    private TaskList taskList;
```

Comment:
I like the naming here! They are short and descriptive enough to explain their roles in the application 👍